src/strategies/seedify-cumulative-voting-power-hodl-staking-farming/index.ts

```diff
@@ -0,0 +1,199 @@
+import { formatUnits } from '@ethersproject/units';
+import { multicall } from '../../utils';
+import { Multicaller } from '../../utils';
+import { strategy as erc20BalanceOfStrategy } from '../erc20-balance-of';
+
+export const author = 'theo6890';
+export const version = '0.1.0';
+
+const sfundStakingAbi = [
+  'function userDeposits(address) external view returns (uint256, uint256, uint256, uint256, uint256, bool)'
+];
+const farmingAbi = [
+  'function userDeposits(address from) external view returns (uint256, uint256, uint256, uint256)'
+];
+const bep20Abi = [
+  'function totalSupply() view returns (uint256)',
+  'function balanceOf(address) view returns (uint256)'
+];
+
+const getStakingBalanceOf = (stakedBalances: any, userIndex: any) => {
+  let sum: number = 0;
+  let balance: any;
+  for (
+    let stakingContractIndex = 0;
+    stakingContractIndex < stakedBalances.length;
+    stakingContractIndex++
+  ) {
+    balance = toDecimals(stakedBalances[stakingContractIndex][userIndex]['0']);
+    sum += balance;
+  }
+  return sum;
+};
+
+const toDecimals = (bigNumber: any) => {
+  return parseFloat(formatUnits(bigNumber.toString(), 18));
+};
+
+const calculateBep20InLPForUser = (
+  lpStaked: any,
+  totalLPSupply: any,
+  totalBep20InPool: any
+) => {
+  lpStaked = toDecimals(lpStaked['0']);
+
+  return (lpStaked / totalLPSupply) * totalBep20InPool;
+};
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+): Promise<Record<string, number>> {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  const createStakingPromises: any = (stakingAddresses: any[]) => {
+    const promises: any = [];
+    for (let i = 0; i < stakingAddresses.length; i++) {
+      promises.push(
+        createPromise(sfundStakingAbi, stakingAddresses[i], 'userDeposits')
+      );
+    }
+    return promises;
+  };
```

Comment:
Hi @Theo6890 we have a limit of 5 calls. this can send unlimited requests, best way to avoid is  to throw an error when there are more than 5 contracts passed into stakingAddresses